tls-client/main.cpp

```diff
@@ -216,23 +219,21 @@ class HelloHTTPS {
 
        /* Start the handshake, the rest will be done in onReceive() */
         mbedtls_printf("Starting the TLS handshake...\r\n");
-        ret = mbedtls_ssl_handshake(&_ssl);
+        do {
+            ret = mbedtls_ssl_handshake(&_ssl);
+        } while (ret != 0 && (ret == MBEDTLS_ERR_SSL_WANT_READ ||
+                ret == MBEDTLS_ERR_SSL_WANT_WRITE));
         if (ret < 0) {
-            if (ret != MBEDTLS_ERR_SSL_WANT_READ &&
-                ret != MBEDTLS_ERR_SSL_WANT_WRITE) {
-                print_mbedtls_error("mbedtls_ssl_handshake", ret);
-                onError(_tcpsocket, -1 );
-            }
+            print_mbedtls_error("mbedtls_ssl_handshake", ret);
```

Comment:
Since ```close``` is moved to the destructor we don't need call to ```onError```. Major reason of not calling ```onError``` is that it misleads in the logs that error has occurred due to Socket.